internal/resources/audit_log_settings.go

```diff
@@ -0,0 +1,323 @@
+}
+
+func NewAuditLogSettings() resource.Resource {
+	return &AuditLogSettings{}
+}
+
+// Metadata returns the audit log settings resource type name.
+func (a *AuditLogSettings) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_auditlogsettings"
+}
+
+// Schema defines the schema for the audit log settings resource.
+func (a *AuditLogSettings) Schema(ctx context.Context, rsc resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = AuditLogSettingsSchema()
+}
+
+// Configure adds the provider configured client to the audit log settings resource.
+func (a *AuditLogSettings) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *ProviderSourceData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	a.Data = data
+}
+
+// AuditLogSettings does not have create endpoint
+// as a workaround, create has same behavior as update
+func (a *AuditLogSettings) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var state providerschema.ClusterAuditSettings
+	diags := req.Plan.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var (
+		organizationId = state.OrganizationId.ValueString()
+		projectId      = state.ProjectId.ValueString()
+		clusterId      = state.ClusterId.ValueString()
+	)
+
+	eventIds := make([]int32, len(state.EnabledEventIDs))
+	for i, event := range state.EnabledEventIDs {
+		eventIds[i] = int32(event.ValueInt64())
+	}
+
+	disabledUsers := make([]api.AuditSettingsDisabledUser, len(state.DisabledUsers))
+	for i, user := range state.DisabledUsers {
+		u := api.AuditSettingsDisabledUser{
+			Domain: user.Domain.ValueStringPointer(),
+			Name:   user.Name.ValueStringPointer(),
+		}
+		disabledUsers[i] = u
+	}
+
+	auditLogUpdateRequest := api.UpdateClusterAuditSettingsRequest{
+		AuditEnabled:    state.AuditEnabled.ValueBool(),
+		EnabledEventIDs: eventIds,
+		DisabledUsers:   disabledUsers,
+	}
+
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/auditLog", a.HostURL, organizationId, projectId, clusterId)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPut, SuccessStatus: http.StatusOK}
+	_, err := a.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		auditLogUpdateRequest,
+		a.Token,
+		nil,
+	)
+
+	if err != nil {
+		resourceNotFound, errString := api.CheckResourceNotFoundError(err)
+		if resourceNotFound {
+			tflog.Info(ctx, "resource doesn't exist in remote server removing resource from state file")
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Error updating audit log settings",
```

Comment:
Are we sure we should mention to the user that it is an update operation, not a create? If so, then we need to have it documented in the provider's documentation.

Reply:
will change to say create